facade/src/main/java/com/redhat/lightblue/migrator/facade/ServiceFacade.java

```diff
@@ -94,6 +105,21 @@ public ServiceFacade(D legacySvc, D lightblueSvc, String implementationName, Pro
 
         timeoutConfiguration = new TimeoutConfiguration(DEFAULT_TIMEOUT_MS, implementationName, this.properties);
 
+        // create ThreadPoolExecutor for all facades
+        synchronized("executor") {
```

Comment:
Actually, what about doing a check if executor is null first?  This way you don't block on synchronization if it is is already set.

```
if (executor == null) {
   synchronized (CONFIG_PREFIX) {
      if (executor == null) {
         ...
      }
   }
}
```